### PR TITLE
fixing notebook

### LINKED ISSUE #13: Rewrite jupyter notebook
In the CLI we rewrote some functions and restructured the backend code --> adjust jupyter notebook.

### PR DESCRIPTION
**Pull request by chStaiger**
The jupyter notebook is adjusted to the new backend.
It replicates each step of the validation pipeline and gives a bit more information.

### CODE REVIEW

**Comment by lagerweijgr** [comment]
For me the changes with jupyter notebook works. I tested with the second example data folder which contains a new version of the files compared to example study1. 
I get the some errors/warnings as when I just run python with this folder, and I like to the explanations of the checks in notebook.
For me the merge is fine and approved.

File: src/validation_pipeline.ipynb [hunk 1]
```diff
@@ -107,7 +108,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "proj_data_path = Path.cwd().parent / \"data\" / \"synthetic_data\"\n",
+    "proj_data_path = Path.cwd().parent / \"data\" / \"synthetic_data_2\"\n",
```
Comment (qubixes) [inline src/validation_pipeline.ipynb:111]: What is synthetic_data_2? Perhaps we can give it a better name? Does the old one still work, some explanation here would be helpful.

File: src/validation_pipeline.ipynb [hunk 2]
```diff
@@ -175,178 +147,134 @@
...
-   "execution_count": null,
-   "id": "b56b73a4-63e4-43dc-a744-c4c9cbc31b4e",
+   "cell_type": "markdown",
+   "id": "3b664d5f-33b4-4658-ab9b-3c6b32256f52",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "if sum(pd.isna(ids[\"host_id\"])):\n",
-    "    print(f\"hosts.csv: There are {sum(pd.isna(ids['host_id']))} empty cells in host_id.\")"
+    "In the current set up we have two identifier columns which can occurr in all of the files:"
```
Comment (qubixes) [inline src/validation_pipeline.ipynb:244]:
```suggestion
    "In the current set up we have two identifier columns which can occur in all of the files:"
```

File: src/validation_pipeline.ipynb [hunk 3]
```diff
@@ -175,178 +147,134 @@
...
   {
    "cell_type": "markdown",
-   "id": "392f8357-6065-4deb-b060-c419f9b2dd55",
+   "id": "965ad2e4-5d56-40f8-b07a-b61c6af53d1c",
    "metadata": {},
    "source": [
-    "### Are all identifiers in the other csv files defined?"
+    "We expect those columns to **not contain any blank cells**.\n",
+    "\n",
+    "The values for the dientifiers are defined in the respcetive columns in *host.csv* and *environment.csv*. In those files the columns must conatin unique values, i.e. the length of the column is equal to the number of unique values found in the column.\n",
```
Comment (qubixes) [inline src/validation_pipeline.ipynb:264]:
```suggestion
    "The values for the identifiers are defined in the respective columns in *host.csv* and *environment.csv*. In those files the columns must contain unique values, i.e. the length of the column is equal to the number of unique values found in the column.\n",
```

File: src/validation_pipeline.ipynb [hunk 4]
```diff
@@ -374,84 +300,88 @@
...
-   "cell_type": "markdown",
-   "id": "3fd0eab6-d107-468e-ba39-96ab4a1599a7",
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "61b6a5d4-4565-4aa5-84ee-6fef14a99dd4",
    "metadata": {},
+   "outputs": [],
    "source": [
-    "## Dependencies between clusters of columns"
+    "# Only for formatting out put\n",
```
Comment (qubixes) [inline src/validation_pipeline.ipynb:332]:
```suggestion
    "# Only for formatting output\n",
```

**Review by qubixes — CHANGES_REQUESTED** [review]
Mostly cosmetic suggestions. The new comments in the notebook are nice to understand exactly what is going on.

File: src/validation_pipeline.ipynb (hunk 1 above)
Comment (chStaiger) [reply src/validation_pipeline.ipynb:111]: Renamed to "test_data"

**Pull request merged**